app/models/like.rb

```diff
@@ -1,2 +1,15 @@
 class Like < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
+
+  after_create :update_post_likes_counter
+  before_destroy :update_likes_before_destroy
+
+  def update_post_likes_counter
+    post.increment!(:likescounter)
+  end
+
+  def update_likes_before_destroy
+    post.decrement!(:likescounter)
+  end
```

Comment:
- [ ] In your `Like` model, consider making the `update_post_likes_counter` and `update_likes_before_destroy` methods private. By marking them as private, you signify that these methods are meant for internal use within the `Like` model and should not be accessed or called from outside the class. This encapsulation helps maintain clean and organized code, preventing unintended usage and potential issues in the future.

```suggestion
private

  def update_post_likes_counter
    post.increment!(:likescounter)
  end

  def update_likes_before_destroy
    post.decrement!(:likescounter)
  end
```